corehq/apps/accounting/static/accounting/js/pricing_table.js

```diff
@@ -1,57 +1,125 @@
+    var PRO = 'pro';
+    var STANDARD = 'standard';
+    var COMMUNITY = 'community';
+
+    var pricingTableModel = function (options) {
+        assertProperties.assert(options, ['editions', 'currentEdition', 'isRenewal', 'startDateAfterMinimum',
+            'isSubscriptionBelowMin', 'nextSubscription']);
+
         'use strict';
         var self = {};
 
-        self.currentEdition = current_edition;
-        self.isRenewal = isRenewal;
-        self.editions = ko.observableArray(_.map(editions, function (edition) {
+        self.currentEdition = options.currentEdition;
+        self.isRenewal = options.isRenewal;
+        self.startDateAfterMinimumSubscription = options.startDateAfterMinimum;
+        self.subscriptionBelowMinimum = options.isSubscriptionBelowMin;
+        self.nextSubscription = options.nextSubscription;
+        self.editions = ko.observableArray(_.map(options.editions, function (edition) {
             return pricingTableEditionModel(edition, self.currentEdition);
         }));
 
-        self.selected_edition = ko.observable(isRenewal ? current_edition : false);
+        self.selected_edition = ko.observable(options.isRenewal ? options.currentEdition : false);
         self.isSubmitVisible = ko.computed(function () {
-            if (isRenewal){
+            if (self.isRenewal){
                 return true;
             }
             return !! self.selected_edition() && !(self.selected_edition() === self.currentEdition);
         });
         self.selectCurrentPlan = function () {
             self.selected_edition(self.currentEdition);
         };
+        self.isDowngrade = function (oldPlan, newPlan) {
+            if (oldPlan === ENTERPRISE) {
+                if (_.contains([ADVANCED, PRO, STANDARD, COMMUNITY], newPlan)) {
+                    return true;
+                }
+            }
+            else if (oldPlan === ADVANCED) {
+                if (_.contains([PRO, STANDARD, COMMUNITY], newPlan)) {
+                    return true;
+                }
+            }
+            else if (oldPlan === PRO) {
+                if (_.contains([STANDARD, COMMUNITY], newPlan)) {
+                    return true;
+                }
+            } else if (oldPlan === STANDARD) {
+                if (newPlan === COMMUNITY) {
+                    return true;
+                }
+            }
+            return false;
+        };
 
         self.form = undefined;
-        self.openDowngradeModal = function(pricingTable, e) {
-            var editionSlugs = _.map(self.editions(), function(e) { return e.slug(); });
+        self.openMinimumSubscriptionModal = function (pricingTable, e) {
             self.form = $(e.currentTarget).closest("form");
-            if (editionSlugs.indexOf(self.selected_edition()) < editionSlugs.indexOf(self.currentEdition)) {
-                var $modal = $("#modal-downgrade");
+
+            var mailto = "<a href=\'mailto:billing-support@dimagi.com\'>billing-support@dimagi.com</a>";
```

Comment:
Code shouldn't have email addresses hard-coded; it's better to pass the `settings.INVOICING_CONTACT_EMAIL` through initial page data.